ddgr

```diff
@@ -1267,8 +1290,15 @@ class DdgCmd(object):
     @no_argument
     def do_next(self):
         # If no results were fetched last time, we have hit the last page already
-        if self._ddg_url._qrycnt == 0:
+        if self._ddg_url._qrycnt == 0 and self.index >= len(self.results):
```

Comment:
What does this additional condition do???

Reply:
I think we will keep having these issues in deciphering stuff if we never get a single line of code comment.

Reply:
You add the check `self.options.pagination` in every other place and then fuse this kind of a condition without any check???

Reply:
Yes, I think this line is the issue with the _huge_ pagination value.

Reply:
I have _hidden_ it by limiting n to 25. If anyone is trying to set n greater than that he is trying to automate stuff. He should patch the code for himself. ;)